services/sidecar/src/simcore_service_sidecar/executor.py

```diff
@@ -215,40 +215,37 @@ async def _pull_image(self):
             config.DOCKER_PASSWORD,
         )
         try:
-            docker_client: aiodocker.Docker = aiodocker.Docker()
-            await self._post_messages(
-                LogType.LOG,
-                f"[sidecar]Pulling {self.task.image['name']}:{self.task.image['tag']}...",
-            )
-            await docker_client.images.pull(
-                docker_image,
-                auth={
-                    "username": config.DOCKER_USER,
-                    "password": config.DOCKER_PASSWORD,
-                },
-            )
-
-            # get integration version
-            image_cfg = await docker_client.images.inspect(docker_image)
-            # NOTE: old services did not have that label
-            if "io.simcore.integration-version" in image_cfg["Config"]["Labels"]:
-                self.integration_version = version.parse(
-                    json.loads(
-                        image_cfg["Config"]["Labels"]["io.simcore.integration-version"]
-                    )["integration-version"]
+            async with aiodocker.Docker() as docker_client:
+                await self._post_messages(
+                    LogType.LOG,
+                    f"[sidecar]Pulling {self.task.image['name']}:{self.task.image['tag']}...",
+                )
+                await docker_client.images.pull(
+                    docker_image,
+                    auth={
+                        "username": config.DOCKER_USER,
+                        "password": config.DOCKER_PASSWORD,
+                    },
                 )
 
+                # get integration version
+                image_cfg = await docker_client.images.inspect(docker_image)
+                # NOTE: old services did not have that label
```

Comment:
MINOR: I wonder if this is shorter and more clear for this type of situations
```python
from contextlib import suppress

#...

with suppress(KeyError):
    self.integration_version = version.parse( 
            json.loads( img_cfg["Config"]["Labels"]["io.simcore.integration-version"] ) 
   )
```

Reply:
it might swallow something that you might need further down the road to understand the nature of a misbehaving test